amazon-kinesis-client-multilang/src/test/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfiguratorTest.java

```diff
@@ -93,6 +94,16 @@ public void testWithLongVariables() {
         assertEquals(config.getShardSyncIntervalMillis(), 500);
     }
 
+    @Test
+    public void testWithInitialPositionInStreamExtended() {
```

Comment:
Nice find! Definitely good to test the exception cases as well.

I've updated the tests. I tried to copy the essence of the unit tests you mentioned over, but didn't exactly copy everything word for word since the 1.0 tests seem to be testing the `KinesisClientLibConfiguration` instead of the `MultiLangDaemonConfiguration`.

So to recap, I believe the unit tests in this test file should now cover these scenarios:

1. When `InitialPositionInStreamExtended` is configured with a timestamp, it sets the AT_TIMESTAMP as well as time in the  `config.getInitialPositionInStreamExtended()`. `config.getInitialPositionInStreamExtended()` seems to already be public or at least package private for `MultiLangDaemonConfiguration`. I don't explicitly see the declaration but I'm assuming it's generated via lomboc https://github.com/awslabs/amazon-kinesis-client/blob/afee84d7da1b5467b1d55f8b675e31b1a412c881/amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/MultiLangDaemonConfiguration.java#L66.
2. Exceptions should be thrown when InitialPositionInStreamExtended is not a long value
3.  Exceptions should be thrown when `InitialPositionInStream` is set to AT_TIMESTAMP
4. TRIM_HORIZON is a valid value for `initialPositionInStream`
5. LATEST is a valid value for `initialPositionInStream`